go/src/koding/config/base_test.go

```diff
@@ -0,0 +1,238 @@
+package config
+
+import (
+	"reflect"
+	"testing"
+	"text/template"
+)
+
+func TestGetEnvironment(t *testing.T) {
+	tests := []struct {
+		Environment string
+		ProvEnv     string
+		Expected    string
+	}{
+		{
+			// 0 //
+			Environment: "development",
+			ProvEnv:     "sandbox",
+			Expected:    "development",
+		},
+		{
+			// 1 //
+			Environment: "production",
+			ProvEnv:     "managed",
+			Expected:    "managed",
+		},
+		{
+			// 2 //
+			Environment: "production",
+			ProvEnv:     "devmanaged",
+			Expected:    "devmanaged",
+		},
+		{
+			// 3 //
+			Environment: "default",
+			ProvEnv:     "sandbox",
+			Expected:    "development",
+		},
+		{
+			// 4 //
+			Environment: "production",
+			ProvEnv:     "production",
+			Expected:    "production",
+		},
+		{
+			// 5 //
+			Environment: "development",
+			ProvEnv:     "devmanaged",
+			Expected:    "devmanaged",
+		},
+	}
+
+	for i, test := range tests {
+		cfg := &Config{
+			Environment: test.Environment,
+		}
+		if gotEnv := cfg.GetEnvironment(test.ProvEnv); gotEnv != test.Expected {
+			t.Errorf("want env = %v; got %v (i:%d)", test.Expected, gotEnv, i)
+		}
+	}
+}
+
+func TestGetBucket(t *testing.T) {
+	var testTmpls = map[string]*template.Template{
+		`buckets.A`:    template.Must(template.New(`buckets.A`).Parse(`{"name":"bucket","region":"reg"}`)),
+		`buckets.Tmpl`: template.Must(template.New(`buckets.Tmpl`).Parse(`{"name":"test-{{.Environment}}","region":"reg-{{.Group}}"}`)),
+		`invalid.Tmpl`: template.Must(template.New(`invalid.Tmpl`).Parse(`{"name":"{{.Missing}}","region":"reg"}`)),
+		`invalid.Json`: template.Must(template.New(`invalid.Json`).Parse(`{"name":bucket,"region":"reg"}`)),
+	}
+
+	tests := []struct {
```

Comment:
just a nitpick - if you replace `[]struct{` with `map[string]struct{` and label each test with a description, you could remove `// 0 //` comments + have a nice description of a test-case (`"this one should fail because...": {` etc.)